src/main/java/io/ballerina/plugins/idea/preloading/TerminatorUnix.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.ballerina.plugins.idea.preloading;
+
+import org.apache.commons.compress.utils.IOUtils;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+
+import static io.ballerina.plugins.idea.BallerinaConstants.BAL_LOG_PREFIX;
+
+/**
+ * Launcher Terminator Implementation for Unix.
+ *
+ * @since 2.0.0
+ */
+public class TerminatorUnix extends Terminator {
+
+    /**
+     * Get find process command.
+     *
+     * @param script absolute path of ballerina file running
+     * @return find process command
+     */
+    private String[] getFindProcessCommand(String script) {
```

Comment:
I think `getFindProcessCommand` is common to both `TerminatorUnix` and `TerminatorMac`. Since, `TerminatorMac` is extending `TerminatorUnix` this method should be marked as protected and has to implement only in `TerminatorUnix`.